---
title: "06_Yoon_Reference"
output: html_document
date: "2023-11-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Calling libraries

```{r Call libraries}
library(tidyverse)
library(Seurat)
```

## Reading files

```{r Read files}
seurat_integrated<-readRDS("../files/scSeq_PA/analysis/seurat_integrated.rds") # Read the scRNA data

#yoon data
yoon_file_list <- list.files("/files/",pattern = "*.csv.gz$",full.names = TRUE)

#cluster genes
cluster_genes <- scan(here::here("deconvolution/download/Yoon_2019/single_cell_cluster_genes.txt"),what=character())
cluster_genes_birey <- readxl::read_xlsx(here::here("deconvolution/download/Birey_scRNAseq_cluster_markers.xlsx")) #actual clusters from Birey's paper (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5805137/)

cluster_genes_birey <- cluster_genes_birey %>% mutate(unique = ! Gene %in% Gene[duplicated(Gene)]) # Look for unique genes

yoon_sc_t <- lapply(yoon_file_list, function(file){
  dt <- data.table::fread(file, nThread = 2)
  t <- data.table::transpose(dt, make.names = "Cell_Label")
  rownames(t) <- colnames(dt)[-1]
  return(t)
})

yoon_dim <- lapply(yoon_sc_t,dim)

yoon_sc_all <- as.data.frame(bind_cols(yoon_sc_t))
rownames(yoon_sc_all) <- rownames(yoon_sc_t[[1]])

saveRDS(yoon_sc_all, file = here::here("yoon_sc_all.rds"))
saveRDS(yoon_dim, file = here::here("yoon_dim.rds"))
```

## Reading files

```{r Read RDS files saved}
# load data from saved RDS files ####
yoon_sc_all <- readRDS(file = here::here("yoon_sc_all.rds"))
yoon_dim <- readRDS(file = here::here("yoon_dim.rds"))

rm(yoon_sc_t)
gc()

```

## Creating Seurat object and performing pre-processing
```{r Create Seurat Object}
yoon_seu <- CreateSeuratObject(counts = yoon_sc_all,
                               project = "yoon",
                               min.cells = 3,
                               min.features = 200)
yoon_seu <- NormalizeData(yoon_seu)
yoon_seu <- FindVariableFeatures(yoon_seu, selection.method = "vst", nfeatures = 2000)
all.genes <- rownames(yoon_seu)
yoon_seu <- ScaleData(yoon_seu, features = all.genes)
yoon_seu <- RunPCA(yoon_seu, features = VariableFeatures(object = yoon_seu))
yoon_seu <- JackStraw(yoon_seu, num.replicate = 100, dims = 40)
yoon_seu <- ScoreJackStraw(yoon_seu, dims = 1:40)
JackStrawPlot(yoon_seu, dims = 1:40,ymax=1)
ElbowPlot(yoon_seu,ndims = 40)

dims <- 1:20
yoon_seu <- FindNeighbors(yoon_seu, dims = dims)

```

## Find clusters at different resolutions
```{r Create Seurat Object}
yoon_seu04 <- FindClusters(yoon_seu, resolution = 0.4)
yoon_seu03 <- FindClusters(yoon_seu, resolution = 0.3)
yoon_seu06 <- FindClusters(yoon_seu, resolution = 0.6)

yoon_seu03 <- RunTSNE(yoon_seu03, dims = dims)
DimPlot(yoon_seu03, reduction = "tsne",pt.size = 1.5,
         cols = c(RColorBrewer::brewer.pal(9,"Set1"),RColorBrewer::brewer.pal(7,"Set2")))

yoon_seu04 <- RunTSNE(yoon_seu04, dims = dims)
DimPlot(yoon_seu04, reduction = "tsne", pt.size = 1.5,
         cols = c(RColorBrewer::brewer.pal(9,"Set1"),RColorBrewer::brewer.pal(7,"Set2")))

yoon_seu06 <- RunTSNE(yoon_seu06, dims = dims)
DimPlot(yoon_seu06, reduction = "tsne",pt.size = 1.5,
        cols = c(RColorBrewer::brewer.pal(9,"Set1"),RColorBrewer::brewer.pal(7,"Set2")))

```


## Find markers at different resolutions
```{r Find markers}
yoon_seu03_markers <- FindAllMarkers(yoon_seu03, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
yoon_seu04_markers <- FindAllMarkers(yoon_seu04, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
yoon_seu06_markers <- FindAllMarkers(yoon_seu06, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)
```

```{r Find genes defined in the original 8 clusters}
genes_per_clust_old <- lapply(paste0("#",1:8), function(cl){
  cluster_genes_birey$Gene[cluster_genes_birey$Cluster == cl & cluster_genes_birey$unique]
})
cluster_genes_birey_unique <- cluster_genes_birey %>% group_by(Gene) %>% summarize(CellTypes = paste(CellType,collapse="_"))
```

## Plotting clusters at different resolutions
```{r seurat 0.3}

yoon_new_markers <- yoon_seu03_markers

genes_per_clust <- lapply((1:length(unique(yoon_new_markers$cluster)))-1, function(cl){
  yoon_new_markers$gene[yoon_new_markers$cluster == cl]
})

comp_markers<-t(sapply(genes_per_clust, function(g){
  check <- sapply(1:8, function(n_old){
    g_old<-genes_per_clust_old[[n_old]]
    p <- sum(g_old %in% g)/length(g_old)
    p
  })
  names(check) <- unique(cluster_genes_birey$CellType)
  check
}))
rownames(comp_markers) <- paste0("#",0:(length(genes_per_clust)-1))

pheatmap::pheatmap(comp_markers,cluster_cols = FALSE,cluster_rows = FALSE)

cluster_names03 <- c("glutamatergicNeuron","IntermediateProgenitor","RadialGlia","Astroglia",
                     "VentralProgenitor","GABAergicNeuron","OPC","ChoroidPlexus")


names(cluster_names03) <- levels(yoon_seu03)
yoon_seu03 <- RenameIdents(yoon_seu03, cluster_names03)
DimPlot(yoon_seu03, reduction = "tsne", label = TRUE, pt.size = 0.5)

yoon_seu03_markers$old_marker <- cluster_genes_birey_unique$CellTypes[match(yoon_seu03_markers$gene,
                                                                            cluster_genes_birey_unique$Gene)]


Seurat::FeaturePlot(yoon_seu03, c("FOXG1","PAX6","EOMES",
                                  "DCX","BCL11B",
                                  "TH","OTX2",                                # midbrain
                                  "GAD1","DLX1","DLX6-AS1","DLX6","SLC32A1")  # interneurons
)



samples<-unlist(lapply(1:3,function(x){rep(paste0("s",x),yoon_dim[[x]][2])}))
samples<-samples[which(colnames(yoon_sc_all) %in% colnames(yoon_seu03[["RNA"]]@counts))]

yoon_data03 <- list(counts = yoon_seu03[["RNA"]]@counts, 
                    featureData = rownames(yoon_seu03[["RNA"]]@counts),
                    phenoData = data.frame(cellname = colnames(yoon_seu03[["RNA"]]@counts), 
                                           cluster = yoon_seu03@active.ident,
                                           sample = samples))


# final files: (comment to protect)
# saveRDS(yoon_seu03, file = here::here("yoon_seurat_object_res03_final.rds"))
# saveRDS(yoon_data03, file = here::here("yoon_data_res03_final.rds"))

```

## Resolution 0.4 ####


```{r seurat 0.4}
yoon_new_markers <- yoon_seu04_markers

genes_per_clust <- lapply((1:length(unique(yoon_new_markers$cluster)))-1, function(cl){
  yoon_new_markers$gene[yoon_new_markers$cluster == cl]
})

comp_markers<-t(sapply(genes_per_clust, function(g){
  check <- sapply(1:8, function(n_old){
    g_old<-genes_per_clust_old[[n_old]]
    p <- sum(g_old %in% g)/length(g_old)
    p
  })
  names(check) <- unique(cluster_genes_birey$CellType)
  check
}))
rownames(comp_markers) <- paste0("#",0:(length(genes_per_clust)-1))

pheatmap::pheatmap(comp_markers,cluster_cols = FALSE,cluster_rows = FALSE)

cluster_names04 <- c("EN_3","EN_1","RG_1","Astro",
                     "glia?","IPC","cc_G2-M",
                     "IN_1","EN_2","Choroid","IN_2"
)

names(cluster_names04) <- levels(yoon_seu04)
yoon_seu04 <- RenameIdents(yoon_seu04, cluster_names04)
DimPlot(yoon_seu04, reduction = "tsne", label = TRUE, pt.size = 0.5)


yoon_seu04_markers$old_marker <- cluster_genes_birey_unique$CellTypes[match(yoon_seu04_markers$gene,cluster_genes_birey_unique$Gene)]
```

##Resolution 0.6
```{r seurat 0.6}

yoon_new_markers <- yoon_seu06_markers

genes_per_clust <- lapply((1:length(unique(yoon_new_markers$cluster)))-1, function(cl){
  yoon_new_markers$gene[yoon_new_markers$cluster == cl]
})

comp_markers<-t(sapply(genes_per_clust, function(g){
  check <- sapply(1:8, function(n_old){
    g_old<-genes_per_clust_old[[n_old]]
    p <- sum(g_old %in% g)/length(g_old)
    p
  })
  names(check) <- unique(cluster_genes_birey$CellType)
  check
}))
rownames(comp_markers) <- paste0("#",0:(length(genes_per_clust)-1))

pheatmap::pheatmap(comp_markers,cluster_cols = FALSE,cluster_rows = FALSE)

cluster_names06 <- c("EN_4","EN_2","RG","glia?","EN_1","Astro","IN_1",
                     "prolif_G2-M","IPC","Astro_Ventral?","EN_3","Choroid","IN_2","other"
)

names(cluster_names06) <- levels(yoon_seu06)
yoon_seu06 <- RenameIdents(yoon_seu06, cluster_names06)
DimPlot(yoon_seu06, reduction = "tsne", label = TRUE, pt.size = 0.5)

yoon_seu06_markers$old_marker <- cluster_genes_birey_unique$CellTypes[match(yoon_seu06_markers$gene,
                                                                            cluster_genes_birey_unique$Gene)]



samples<-unlist(lapply(1:3,function(x){rep(paste0("s",x),yoon_dim[[x]][2])}))
samples<-samples[which(colnames(yoon_sc_all) %in% colnames(yoon_seu06[["RNA"]]@counts))]

yoon_data06 <- list(counts = yoon_seu06[["RNA"]]@counts, 
                    featureData = rownames(yoon_seu06[["RNA"]]@counts),
                    phenoData = data.frame(cellname = colnames(yoon_seu06[["RNA"]]@counts), 
                                           cluster = yoon_seu06@active.ident,
                                           sample = samples))


# final files
# saveRDS(yoon_seu06, file = here::here("yoon_seurat_object_res06_final.rds"))
# saveRDS(yoon_data06, file = here::here("yoon_data_res06_final.rds"))
# 

```

## Mapping this reference data set with scRNA data

```{r Calling the ref files}
# 1) Converting query data to single cell experiment
scRNA.sce <- as.SingleCellExperiment(seurat_integrated)


# 2) Yoon dataset

Yoon_reference<-readRDS("yoon_seurat_object_res03_final.rds")

Yoon_reference <- subset(Yoon_reference,cells = WhichCells(Yoon_reference, downsample = 200))

Yoon_reference <- SCTransform(Yoon_reference, ncells = 3000, verbose = FALSE, method = "poisson") %>%
  RunPCA(verbose = FALSE) %>%
  RunUMAP(dims = 1:30)

AR<- as.SingleCellExperiment(Yoon_reference)

pred.hneurons<-SingleR(test = scRNA.sce,ref = AR,labels = AR$ident)


cdata = cbind(colData(scRNA.sce), 
              data.frame(HumanNeurons.label = pred.hneurons$pruned.labels))

rescounts = table(pred.hneurons$pruned.labels, useNA="ifany")
othercells = names(rescounts)[rescounts<100]

cdata$Neurons100.label = pred.hneurons$pruned.labels
cdata$Neurons100.label[cdata$Neurons100.label %in%othercells] = "other"

Neuronpalette = colorRampPalette(brewer.pal(12, "Paired"))(length(unique(cdata$Neurons100.label)))
names(Neuronpalette) = unique(cdata$Neurons100.label)
Neuronpalette["other"] = "gray30"

colData(scRNA.sce) = cdata

# Adding reduced dims to the seurat converted object

umap<-Embeddings(seurat_integrated,"umap")[,]
reducedDim(scRNA.sce, "UMAP")<-umap

pca<-Embeddings(seurat_integrated,"pca")[,]
reducedDim(scRNA.sce, "PCA")<-pca

tsne<-Embeddings(seurat_integrated,"tsne")[,]
reducedDim(scRNA.sce, "TSNE")<-tsne


colData(scRNA.sce)$UMAP1<-data.frame(reducedDim(scRNA.sce)[,1])
colData(scRNA.sce)$UMAP2<-data.frame(reducedDim(scRNA.sce)[,2])

scRNA.sce$UMAP1<-as.numeric(scRNA.sce$UMAP1$reducedDim.scRNA.sce....1.)
scRNA.sce$UMAP2<-as.numeric(scRNA.sce$UMAP2$reducedDim.scRNA.sce....2.)


plotScoreHeatmap(pred.hneurons)

colsname<-unname(alphabet())[1:16]

pdf("Yoon_Reference_0.3.pdf")
gg<-scater::plotReducedDim(scRNA.sce, 'UMAP', colour_by = 'Neurons100.label') 
gg+scale_fill_manual(values=c(glutamatergicNeuron="red", OPC="green",ChoroidPlexus="blue",IntermediateProgenitor="pink",
                           VentralProgenitor="brown",Astroglia="purple",GABAergicNeuron="orange",RadialGlia="yellow",other="grey"),
                    aesthetics="colour")
dev.off()


pdf("Yoon_Reference_0.6.pdf")
gg<-scater::plotReducedDim(scRNA.sce, 'UMAP', colour_by = 'Neurons100.label') 
gg + scale_fill_manual(values=colsname, aesthetics="colour", name="Celltype")
dev.off()

pdf("Yoon_Reference_0.6.pdf")
cowplot::plot_grid(
  scater::plotReducedDim(scRNA.sce, 'UMAP', colour_by = 'Neurons100.label') 
  + scale_fill_manual(values=colsname, aesthetics="colour", name="Celltype"),
  scater::plotReducedDim(scRNA.sce, 'UMAP', colour_by = 'seurat_clusters', text_by = 'seurat_clusters')
  + scale_fill_manual(values=colsname,aesthetics="colour",  name="Clusters"))
dev.off()

pdf("Yoon_Reference_sc.pdf")
  scater::plotReducedDim(scRNA.sce, 'UMAP', colour_by = 'Neurons100.label') 
  scale_fill_manual(values=colsname, aesthetics="colour", name="Celltype")
dev.off()
```
