<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A Chiocchetti" />

<meta name="date" content="2024-01-09" />

<title>DEX Analysis</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">QPRT-inhibition scRNA organoids</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://chiocchettilab.uni-frankfurt.de">Chiocchettilab</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">DEX Analysis</h1>
<h4 class="author">A Chiocchetti</h4>
<h4 class="date">2024-01-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>files/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240103code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240103)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240103code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240103)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscRNAPAQPRTtreef7ee227768eadf71185ea14617a3ddf4ca392ee0targetblankf7ee227a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/tree/f7ee227768eadf71185ea14617a3ddf4ca392ee0" target="_blank">f7ee227</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomKJPMolgenLabscRNAPAQPRTtreef7ee227768eadf71185ea14617a3ddf4ca392ee0targetblankf7ee227a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/tree/f7ee227768eadf71185ea14617a3ddf4ca392ee0" target="_blank">f7ee227</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rprofile
    Ignored:    .Rproj.user/
    Ignored:    output/05_Bulk_Preprocessing/
    Ignored:    output/07_Bulk_Deconv/

Untracked files:
    Untracked:  .bashrc
    Untracked:  .nuvolos/
    Untracked:  data/Bulk_dds_matrix.rds
    Untracked:  data/Bulkdata_Countmatrix.rds
    Untracked:  data/HumDev/
    Untracked:  data/Kanton/
    Untracked:  data/KantonGeneList/
    Untracked:  data/Knoblich/
    Untracked:  data/QC_data.filt.rds
    Untracked:  data/QC_dataClustered.filt.rds
    Untracked:  data/QC_dataClustered.filtTyped.rds
    Untracked:  data/QC_dataClustered.filtTypedTraj.rds
    Untracked:  data/QC_dataFinal.rds
    Untracked:  data/Yoon/
    Untracked:  data/alldata.rds

Unstaged changes:
    Modified:   .gitignore
    Modified:   output/02_Clustering/Cluster_markers.xls
    Modified:   output/02_Clustering/Clusters_GO_termsCluster.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_allcells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_0cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_10cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_11cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_12cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_13cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_1cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_2cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_3cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_4cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_5cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_6cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_7cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_8cells.xlsx
    Modified:   output/04_Biostat/Biostat_DEG_cluster_9cells.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_all.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_0.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_1.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_10.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_11.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_12.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_13.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_2.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_3.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_4.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_5.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_6.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_7.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_8.xlsx
    Modified:   output/04_Biostat/Biostat_GO_termDEG_cluster_9.xlsx
    Modified:   output/04_Biostat/Biostat_HumdevCTFreq.xslsx
    Modified:   output/04_Biostat/Biostat_clusterFreq.xslsx
    Modified:   workflowHelper.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/06_DEX_Bulk.Rmd</code>) and HTML
(<code>docs/06_DEX_Bulk.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/f7ee227768eadf71185ea14617a3ddf4ca392ee0/analysis/06_DEX_Bulk.Rmd" target="_blank">f7ee227</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-11
</td>
<td>
wflow_publish(c("./analysis/<em>.Rmd", "./code/</em>", "./docs/*"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KJPMolgenLab/scRNA_PA_QPRT/9044a86b031be6dc5e252fe7adc80086d2294d09/docs/06_DEX_Bulk.html" target="_blank">9044a86</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/f206c289f1dda0185da1dcbfdc067a262aba4279/analysis/06_DEX_Bulk.Rmd" target="_blank">f206c28</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-11
</td>
<td>
wflow_publish(c("./analysis/<em>.Rmd", "./code/</em>", "./docs/*"))
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/e6ee176f53438a82428b0756717d7c916446bfae/analysis/06_DEX_Bulk.Rmd" target="_blank">e6ee176</a>
</td>
<td>
achiocch
</td>
<td>
2024-01-10
</td>
<td>
add DEX Bulk analysis plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/133d5ee7a404a20939fdeb12dceb408cedd3dd5c/analysis/06_DEX_Bulk.Rmd" target="_blank">133d5ee</a>
</td>
<td>
achiocch
</td>
<td>
2024-01-10
</td>
<td>
add DEX Bulk analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/bca1a73fcaee6c918ab64bd157773ebacb6c9bb4/analysis/06_DEX_Bulk.Rmd" target="_blank">bca1a73</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-09
</td>
<td>
intermediate stage bulk seq
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(knitr)
library(RCurl) 
library(DESeq2)</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:utils&#39;:

    findMatches</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MatrixGenerics&#39;:

    rowMedians</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    anyMissing, rowMedians</code></pre>
<pre class="r"><code>library(RColorBrewer)
library(glmpca)
library(ggplot2)
library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ lubridate 1.9.3     ✔ tibble    3.2.1
✔ purrr     1.0.2     ✔ tidyr     1.3.0</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ lubridate::%within%() masks IRanges::%within%()
✖ dplyr::collapse()     masks IRanges::collapse()
✖ dplyr::combine()      masks Biobase::combine(), BiocGenerics::combine()
✖ tidyr::complete()     masks RCurl::complete()
✖ dplyr::count()        masks matrixStats::count()
✖ dplyr::desc()         masks IRanges::desc()
✖ tidyr::expand()       masks S4Vectors::expand()
✖ dplyr::filter()       masks stats::filter()
✖ dplyr::first()        masks S4Vectors::first()
✖ dplyr::lag()          masks stats::lag()
✖ ggplot2::Position()   masks BiocGenerics::Position(), base::Position()
✖ purrr::reduce()       masks GenomicRanges::reduce(), IRanges::reduce()
✖ dplyr::rename()       masks S4Vectors::rename()
✖ lubridate::second()   masks S4Vectors::second()
✖ lubridate::second&lt;-() masks S4Vectors::second&lt;-()
✖ dplyr::slice()        masks IRanges::slice()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(pheatmap)
library(dendextend)</code></pre>
<pre><code>
---------------------
Welcome to dendextend version 1.17.1
Type citation(&#39;dendextend&#39;) for how to cite the package.

Type browseVignettes(package = &#39;dendextend&#39;) for the package vignette.
The github page is: https://github.com/talgalili/dendextend/

Suggestions and bug-reports can be submitted at: https://github.com/talgalili/dendextend/issues
You may ask questions at stackoverflow, use the r and dendextend tags: 
     https://stackoverflow.com/questions/tagged/dendextend

    To suppress this message use:  suppressPackageStartupMessages(library(dendextend))
---------------------


Attaching package: &#39;dendextend&#39;

The following object is masked from &#39;package:stats&#39;:

    cutree</code></pre>
<pre class="r"><code>library(gprofiler2)</code></pre>
<pre class="r"><code>options(stringsAsFactors = F)
set.seed(157434538)
reanalyze=F
cutoff = 0.05 # pavalue cutoff for DEG
padj_cu=FALSE # use adjusted pvalue 
Dark8 = RColorBrewer::brewer.pal(8, &quot;Dark2&quot;)</code></pre>
<pre class="r"><code>output= paste0(&quot;output/06_Bulk_DEX/&quot;)
dir.create(output, showWarnings = F)
source(&quot;code/custom_functions.R&quot;)</code></pre>
<pre><code>Loading required package: kableExtra</code></pre>
<pre><code>
Attaching package: &#39;kableExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    group_rows</code></pre>
<pre><code>Loading required package: compareGroups</code></pre>
<pre class="r"><code>ddsMat &lt;- readRDS(&quot;data/Bulk_dds_matrix.rds&quot;)
SampleInfo = as.data.frame(colData(ddsMat))
log_2cpm=log2(counts(ddsMat, normalize=T)+1)</code></pre>
<pre class="r"><code>## comparisons against 0MM PA
PA=levels(SampleInfo$PAmM)
Days=c(sort(unique(SampleInfo$Day)))

p = 0
d = Days[1]
target = &quot;PAmM&quot;
randomeffect = &quot;Batch&quot;

geneinfo = rowData(ddsMat) %&gt;% as.data.frame()
geneinfo$entrez_gene = rownames(geneinfo)
geneinfo$hgnc[is.na(geneinfo$hgnc)] &lt;- geneinfo$entrez_gene[is.na(geneinfo$hgnc)]

for(d in Days){
  Dayfilter = SampleInfo$Day == d
  Set = rownames(SampleInfo)[Dayfilter]
  lab = paste(&quot;restab&quot;, &quot;Day&quot;, d,&quot;0vs5mM_PA&quot;,sep=&quot;_&quot;)
  print(lab)
  res &lt;- comparison(dds_object = ddsMat, 
                    samples = Set, 
                    target = target, 
                    randomeffect = randomeffect)
  res$HGNC = geneinfo[match(as.character(rownames(res)), as.character(rownames(geneinfo))), &quot;hgnc&quot;]
  assign(lab, res)
}</code></pre>
<pre><code>[1] &quot;restab_Day_5_0vs5mM_PA&quot;</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre><code>
Attaching package: &#39;limma&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:DESeq2&#39;:

    plotMA</code></pre>
<pre><code>The following object is masked from &#39;package:BiocGenerics&#39;:

    plotMA</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>Removing intercept from test coefficients</code></pre>
<pre><code>[1] &quot;restab_Day_10_0vs5mM_PA&quot;</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>Removing intercept from test coefficients</code></pre>
<pre><code>[1] &quot;restab_Day_17_0vs5mM_PA&quot;</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>Removing intercept from test coefficients</code></pre>
<pre><code>[1] &quot;restab_Day_40_0vs5mM_PA&quot;</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>Removing intercept from test coefficients</code></pre>
<pre><code>[1] &quot;restab_Day_80_0vs5mM_PA&quot;</code></pre>
<pre><code>Warning: Zero sample variances detected, have been offset away from zero</code></pre>
<pre><code>Removing intercept from test coefficients</code></pre>
<pre class="r"><code>#Lists elements are written to individual worksheets, using list names as sheet names if available
l &lt;- list(mget(apropos(&quot;restab_&quot;)))[[1]]
names(l) = gsub(&quot;restab_&quot;, &quot;&quot;, names(l)) %&gt;% gsub(&quot;Day_5&quot;, &quot;Day_05&quot;,.)
l = l[sort(names(l))]
openxlsx2::write_xlsx(l, file=paste0(output, &quot;06_Bulk_DEX_LFC_results.xlsx&quot;), row_names=T )</code></pre>
<pre class="r"><code>geneuniverses &lt;- rownames(log_2cpm)
GOtermlist &lt;- list()

for (i in apropos(&quot;restab_&quot;)){
  restab &lt;- get(i)
  if(padj_cu){
    GOtermlist[[paste0(i, &quot;_all&quot;)]] &lt;- rownames(restab)[restab$adj.P.Val &lt; cutoff]
    GOtermlist[[paste0(i, &quot;_up&quot;)]] &lt;- rownames(restab)[restab$adj.P.Val &lt; cutoff  &amp; restab$logFC &gt;0]
    GOtermlist[[paste0(i, &quot;_dwn&quot;)]] &lt;- rownames(restab)[restab$adj.P.Val &lt; cutoff  &amp; restab$logFC &lt; 0]
  }else{
    GOtermlist[[paste0(i, &quot;_all&quot;)]] &lt;- rownames(restab)[restab$P.Value &lt; cutoff]
    GOtermlist[[paste0(i, &quot;_up&quot;)]] &lt;- rownames(restab)[restab$P.Value &lt; cutoff  &amp; restab$logFC &gt;0]
    GOtermlist[[paste0(i, &quot;_dwn&quot;)]] &lt;- rownames(restab)[restab$P.Value &lt; cutoff  &amp; restab$logFC &lt; 0]
  }
}

names(GOtermlist) = gsub(&quot;restab_Day&quot;, &quot;Day&quot;, names(GOtermlist))
names(GOtermlist) = gsub(&quot;Day_5&quot;, &quot;Day_05&quot;, names(GOtermlist))
GOtermlist = GOtermlist[sort(names(GOtermlist))]

resGO = getGOresults(GOtermlist, genereference = NULL, evcodes = T)

openxlsx::write.xlsx(file = paste0(output, &quot;/06_Bulk_DEX_GO_results.xlsx&quot;), as.data.frame(resGO$result))</code></pre>
<pre class="r"><code>plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set),]
  if(nrow(dat)&gt;0){
    plotlist[[set]]&lt;-GOplot(dat, N = 10, Title = set)
  }else{
    text=&quot;no signifiance&quot;
    plotlist[[set]]&lt;-ggplot() + annotate(&quot;text&quot;, x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p &lt;- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="figure/06_DEX_Bulk.Rmd/unnamed-chunk-4-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/9044a86b031be6dc5e252fe7adc80086d2294d09/docs/figure/06_DEX_Bulk.Rmd/unnamed-chunk-4-1.png" target="_blank">9044a86</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=paste0(output, &quot;/06_Bulk_DEX_top10all_GOplot.svg&quot;), plot = p)</code></pre>
<pre><code>Saving 15 x 20 in image</code></pre>
<pre class="r"><code>plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set &amp; resGO$result$source == &quot;GO:BP&quot;),]
  if(nrow(dat)&gt;0){
    plotlist[[set]]&lt;-GOplot(dat, N = 10, Title = set)
  }else{
    text=&quot;no signifiance&quot;
    plotlist[[set]]&lt;-ggplot() + annotate(&quot;text&quot;, x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p &lt;- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p</code></pre>
<p><img src="figure/06_DEX_Bulk.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/9044a86b031be6dc5e252fe7adc80086d2294d09/docs/figure/06_DEX_Bulk.Rmd/unnamed-chunk-5-1.png" target="_blank">9044a86</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=paste0(output, &quot;/06_Bulk_DEX_top10_BP_GOplot.svg&quot;), plot = p)</code></pre>
<pre><code>Saving 15 x 20 in image</code></pre>
<pre class="r"><code>plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set &amp; resGO$result$source == &quot;TF&quot;),]
  if(nrow(dat)&gt;0){
    plotlist[[set]]&lt;-GOplot(dat, N = 10, Title = set)
  }else{
    text=&quot;no signifiance&quot;
    plotlist[[set]]&lt;-ggplot() + annotate(&quot;text&quot;, x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p &lt;- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p</code></pre>
<p><img src="figure/06_DEX_Bulk.Rmd/unnamed-chunk-6-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/KJPMolgenLab/scRNA_PA_QPRT/blob/9044a86b031be6dc5e252fe7adc80086d2294d09/docs/figure/06_DEX_Bulk.Rmd/unnamed-chunk-6-1.png" target="_blank">9044a86</a>
</td>
<td>
Andreas Chiocchetti
</td>
<td>
2024-01-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggsave(file=paste0(output, &quot;/06_Bulk_DEX_top10_TF_GOplot.svg&quot;), plot = p)</code></pre>
<pre><code>Saving 15 x 20 in image</code></pre>
<pre class="r"><code>topgenes=c()
for(r in apropos(&quot;restab_&quot;)){
  res &lt;- get(r)
  topgenes &lt;- c(topgenes, res$HGNC[1:10])
}

topgenes &lt;- unique(topgenes)
topgenes &lt;- sort(topgenes)

plotDEX= function(geneToPlot){
  data.frame(reads = log_2cpm[which(geneinfo$hgnc==geneToPlot),], Group = SampleInfo$PAmM, Day = SampleInfo$Day) %&gt;%
    ggplot(aes(x = Group, y = reads, fill = Group)) +
    geom_boxplot() +
    theme_classic() + facet_wrap(~Day, nrow  = 1) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
    labs(x = &quot;Day&quot;, y = &quot;log2(CPM+1)&quot;, title = geneToPlot) -&gt; p
  return(p)
}


plotlist=list()
for(g in topgenes){
  p = plotDEX(g)
  plotlist[[g]] &lt;- p
}

pcomb = ggpubr::ggarrange(plotlist=plotlist, nrow = 5, ncol = 4, common.legend = T, heights = c(3,3,3,3,3), widths = c(3,3,3,3))
ggpubr::ggexport(filename = paste0(output, &quot;/06_Bulk_DEX_Siggenes_boxplots.pdf&quot;), plotlist = pcomb, width = 14, height = 21, dpi = 300)</code></pre>
<pre><code>file saved to output/06_Bulk_DEX//06_Bulk_DEX_Siggenes_boxplots.pdf</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] limma_3.56.2                compareGroups_4.7.2        
 [3] kableExtra_1.3.4            gprofiler2_0.2.2           
 [5] dendextend_1.17.1           pheatmap_1.0.12            
 [7] lubridate_1.9.3             forcats_1.0.0              
 [9] stringr_1.5.1               dplyr_1.1.4                
[11] purrr_1.0.2                 readr_2.1.4                
[13] tidyr_1.3.0                 tibble_3.2.1               
[15] tidyverse_2.0.0             ggplot2_3.4.4              
[17] glmpca_0.2.0                RColorBrewer_1.1-3         
[19] DESeq2_1.40.2               SummarizedExperiment_1.32.0
[21] Biobase_2.62.0              MatrixGenerics_1.14.0      
[23] matrixStats_1.1.0           GenomicRanges_1.54.1       
[25] GenomeInfoDb_1.38.1         IRanges_2.36.0             
[27] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[29] RCurl_1.98-1.13             knitr_1.45                 
[31] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] splines_4.3.1           later_1.3.2             bitops_1.0-7           
  [4] HardyWeinberg_1.7.5     rpart_4.1.23            lifecycle_1.0.4        
  [7] rstatix_0.7.2           rprojroot_2.0.4         processx_3.8.3         
 [10] Rsolnp_1.16             lattice_0.22-5          MASS_7.3-60            
 [13] backports_1.4.1         magrittr_2.0.3          openxlsx_4.2.5.2       
 [16] plotly_4.10.3           sass_0.4.8              rmarkdown_2.25         
 [19] jquerylib_0.1.4         yaml_2.3.8              httpuv_1.6.13          
 [22] zip_2.3.0               askpass_1.2.0           cowplot_1.1.2          
 [25] minqa_1.2.6             abind_1.4-5             zlibbioc_1.48.0        
 [28] rvest_1.0.3             nnet_7.3-19             openxlsx2_1.2          
 [31] git2r_0.33.0            gdtools_0.3.5           GenomeInfoDbData_1.2.11
 [34] crul_1.4.0              svglite_2.1.3           codetools_0.2-19       
 [37] DelayedArray_0.28.0     xml2_1.3.6              tidyselect_1.2.0       
 [40] shape_1.4.6             farver_2.1.1            httpcode_0.3.0         
 [43] lme4_1.1-35.1           viridis_0.6.4           webshot_0.5.5          
 [46] jsonlite_1.8.8          mitml_0.4-5             ellipsis_0.3.2         
 [49] survival_3.5-7          iterators_1.0.14        systemfonts_1.0.5      
 [52] foreach_1.5.2           tools_4.3.1             chron_2.3-61           
 [55] ragg_1.2.7              Rcpp_1.0.11             glue_1.6.2             
 [58] gridExtra_2.3           pan_1.9                 SparseArray_1.2.2      
 [61] xfun_0.41               withr_2.5.2             fastmap_1.1.1          
 [64] boot_1.3-28.1           fansi_1.0.6             openssl_2.1.1          
 [67] callr_3.7.3             digest_0.6.33           truncnorm_1.0-9        
 [70] timechange_0.2.0        R6_2.5.1                mime_0.12              
 [73] mice_3.16.0             textshaping_0.3.7       colorspace_2.1-0       
 [76] utf8_1.2.4              generics_0.1.3          fontLiberation_0.1.0   
 [79] data.table_1.14.10      httr_1.4.7              htmlwidgets_1.6.4      
 [82] S4Arrays_1.2.0          whisker_0.4.1           pkgconfig_2.0.3        
 [85] gtable_0.3.4            XVector_0.42.0          htmltools_0.5.7        
 [88] carData_3.0-5           fontBitstreamVera_0.1.1 scales_1.3.0           
 [91] rstudioapi_0.15.0       tzdb_0.4.0              uuid_1.1-1             
 [94] nlme_3.1-164            curl_5.2.0              nloptr_2.0.3           
 [97] cachem_1.0.8            flextable_0.9.4         parallel_4.3.1         
[100] pillar_1.9.0            grid_4.3.1              vctrs_0.6.5            
[103] ggpubr_0.6.0            promises_1.2.1          car_3.1-2              
[106] jomo_2.7-6              xtable_1.8-4            evaluate_0.23          
[109] cli_3.6.2               locfit_1.5-9.8          compiler_4.3.1         
[112] rlang_1.1.2             crayon_1.5.2            ggsignif_0.6.4         
[115] labeling_0.4.3          ps_1.7.5                getPass_0.2-4          
[118] fs_1.6.3                writexl_1.4.2           stringi_1.8.3          
[121] viridisLite_0.4.2       BiocParallel_1.34.2     munsell_0.5.0          
[124] lazyeval_0.2.2          glmnet_4.1-8            fontquiver_0.2.1       
[127] Matrix_1.6-4            hms_1.1.3               gfonts_0.2.0           
[130] statmod_1.5.0           shiny_1.8.0             highr_0.10             
[133] broom_1.0.5             bslib_0.6.1             officer_0.6.3          </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
