---
title: "DEX Analysis"
author: "A Chiocchetti"
date: "2024-01-09"
output:
  html_document:
    theme: cerulean
    toc: yes
    toc_float:
      collapsed: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
```

```{r}
library(knitr)
library(RCurl) 
library(DESeq2)
library(RColorBrewer)
library(glmpca)
library(ggplot2)
library(tidyverse)
library(pheatmap)
library(dendextend)
library(gprofiler2)
```
```{r}
options(stringsAsFactors = F)
set.seed(157434538)
reanalyze=F
cutoff = 0.05 # pavalue cutoff for DEG
padj_cu=FALSE # use adjusted pvalue 
Dark8 = RColorBrewer::brewer.pal(8, "Dark2")
```

```{r getdata}
output= paste0("output/06_Bulk_DEX/")
dir.create(output, showWarnings = F)
source("code/custom_functions.R")
ddsMat <- readRDS("data/Bulk_dds_matrix.rds")
chr16p11_genes <- openxlsx::read.xlsx("data/chr16p11_orig.xlsx")
Kynurenine_genes <- openxlsx::read.xlsx("data/Kynurenine_genes_orig.xlsx")

SampleInfo = as.data.frame(colData(ddsMat))
log_2cpm=log2(counts(ddsMat, normalize=T)+1)

hgnc=gconvert(query=as.numeric(rownames(ddsMat)), 
              organism = "hsapiens", 
              numeric_ns = "ENTREZGENE_ACC",
              target = "HGNC")

```

# Differential gene expression 

```{r analyze DEX}
## comparisons against 0MM PA
PA=levels(SampleInfo$PAmM)
Days=c(sort(unique(SampleInfo$Day)))

p = 0
d = Days[1]
target = "PAmM"
randomeffect = "Batch"

geneinfo = rowData(ddsMat) %>% as.data.frame()
geneinfo$entrez_gene = rownames(geneinfo)
geneinfo$hgnc[is.na(geneinfo$hgnc)] <- geneinfo$entrez_gene[is.na(geneinfo$hgnc)]

for(d in Days){
  Dayfilter = SampleInfo$Day == d
  Set = rownames(SampleInfo)[Dayfilter]
  lab = paste("restab", "Day", d,"0vs5mM_PA",sep="_")
  print(lab)
  res <- comparison(dds_object = ddsMat, 
                    samples = Set, 
                    target = target, 
                    randomeffect = randomeffect)
  res$HGNC = geneinfo[match(as.character(rownames(res)), as.character(rownames(geneinfo))), "hgnc"]
  assign(lab, res)
}

#Lists elements are written to individual worksheets, using list names as sheet names if available
l <- list(mget(apropos("restab_")))[[1]]
names(l) = gsub("restab_", "", names(l)) %>% gsub("Day_5", "Day_05",.)
l = l[sort(names(l))]
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_DEX_LFC_results.xlsx"), row_names=T )
```

## GO terms
```{r, fig.width=15, fig.height=20}
geneuniverses <- rownames(log_2cpm)
GOtermlist <- list()

for (i in apropos("restab_")){
  restab <- get(i)
  if(padj_cu){
    GOtermlist[[paste0(i, "_all")]] <- rownames(restab)[restab$adj.P.Val < cutoff]
    GOtermlist[[paste0(i, "_up")]] <- rownames(restab)[restab$adj.P.Val < cutoff  & restab$logFC >0]
    GOtermlist[[paste0(i, "_dwn")]] <- rownames(restab)[restab$adj.P.Val < cutoff  & restab$logFC < 0]
  }else{
    GOtermlist[[paste0(i, "_all")]] <- rownames(restab)[restab$P.Value < cutoff]
    GOtermlist[[paste0(i, "_up")]] <- rownames(restab)[restab$P.Value < cutoff  & restab$logFC >0]
    GOtermlist[[paste0(i, "_dwn")]] <- rownames(restab)[restab$P.Value < cutoff  & restab$logFC < 0]
  }
}

names(GOtermlist) = gsub("restab_Day", "Day", names(GOtermlist))
names(GOtermlist) = gsub("Day_5", "Day_05", names(GOtermlist))
GOtermlist = GOtermlist[sort(names(GOtermlist))]

resGO = getGOresults(GOtermlist, genereference = NULL, evcodes = T)

openxlsx::write.xlsx(file = paste0(output, "/06_Bulk_DEX_GO_results.xlsx"), as.data.frame(resGO$result))

```

```{r, fig.width=15, fig.height=20}

plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set, xlim=NA)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DEX_top10all_GOplot.svg"), plot = p)
```

```{r, fig.width=15, fig.height=20}
plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set & resGO$result$source == "GO:BP"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set, xlim=NA)
  }else{
    text="no signifiance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DEX_top10_BP_GOplot.svg"), plot = p)
```


```{r, fig.width=15, fig.height=20}
plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set & resGO$result$source == "TF"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set, xlim=NA)
  }else{
    text="no signifiance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DEX_top10_TF_GOplot.svg"), plot = p)
```

## Trajectories over time top genes per comparison

```{r plot top 10 DEX genes, fig.width=14, fig.height=21}

topgenes=c()
for(r in apropos("restab_")){
  res <- get(r)
  topgenes <- c(topgenes, res$HGNC[1:10])
}

topgenes <- unique(topgenes)
topgenes <- sort(topgenes)

plotDEX= function(geneToPlot){
  data.frame(reads = log_2cpm[which(geneinfo$hgnc==geneToPlot),], Group = SampleInfo$PAmM, Day = SampleInfo$Day) %>%
    ggplot(aes(x = Group, y = reads, fill = Group)) +
    geom_boxplot() +
    theme_classic() + facet_wrap(~Day, nrow  = 1) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
    labs(x = "Day", y = "log2(CPM+1)", title = geneToPlot) -> p
  return(p)
}


plotlist=list()
for(g in topgenes){
  p = plotDEX(g)
  plotlist[[g]] <- p
}

pcomb = ggpubr::ggarrange(plotlist=plotlist, nrow = 5, ncol = 4, common.legend = T, heights = c(3,3,3,3,3), widths = c(3,3,3,3))
pcomb
ggpubr::ggexport(filename = paste0(output, "/06_Bulk_DEX_Siggenes_boxplots.pdf"), plotlist = pcomb, width = 14, height = 21, dpi = 300)

dataplot <- log_2cpm[which(geneinfo$hgnc%in%topgenes),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$PAmM = SampleInfo[dataplot$Sample, "PAmM"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads,col=PAmM, fill = PAmM)) +
    geom_smooth() +
    theme_classic() + facet_wrap(~hgnc) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
    labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_Siggenes_lineplots.svg"), p)
```
## Kynurenine genes over time

```{r, fig.width=7, fig.height=10}
dataplot <- log_2cpm[which(geneinfo$hgnc%in%Kynurenine_genes$Gene_symbol),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$PAmM = SampleInfo[dataplot$Sample, "PAmM"]
dataplot$Batch = SampleInfo[dataplot$Sample, "Batch"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads,col=PAmM, fill = PAmM)) +
    geom_smooth() +
    theme_classic() + facet_wrap(~hgnc) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
    labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_Kynurenine_lineplots.svg"), p)

results<-data.frame(matrix(ncol=2, nrow=0))
for(g in unique(dataplot$hgnc)){
  subdat<-dataplot[dataplot$hgnc == g, ]
  res <- lme4::glmer(reads~Day*PAmM+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[4]))
}

openxlsx2::write_xlsx(file=paste0(output, "/06_Bulk_DEX_Kynurenine_stat_DayxmM.xlsx"), results)

colnames(results) <- c("gene", "Pval")
display_tab(results)


```

```{r, fig.width=7, fig.height=10}
dataplot <- log_2cpm[which(geneinfo$hgnc%in%chr16p11_genes$Gene_symbol),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$PAmM = SampleInfo[dataplot$Sample, "PAmM"]
dataplot$Batch = SampleInfo[dataplot$Sample, "Batch"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads,col=PAmM, fill = PAmM)) +
    geom_smooth() +
    theme_classic() + facet_wrap(~hgnc) +
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
    labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_16p11_lineplots.svg"), p)

results<-data.frame(matrix(ncol=2, nrow=0))

for(g in unique(dataplot$hgnc)){
  subdat<-dataplot[dataplot$hgnc == g, ]
  res <- lme4::glmer(reads~Day*PAmM+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[4]))
}


openxlsx2::write_xlsx(file=paste0(output, "/06_Bulk_DEX_chr16p11_stat_DayxmM.xlsx"), results)
colnames(results) <- c("gene", "Pval")
display_tab(results)

```
