---
title: "DEX Analysis"
author: "A Chiocchetti"
date: "2024-01-09"
output:
  html_document:
    theme: cerulean
    toc: yes
    toc_float:
      collapsed: true
---

# DEX Analysis 

```{r setup, eval=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
```

```{r}
library(knitr)
library(RCurl) 
library(DESeq2)
library(RColorBrewer)
library(glmpca)
library(ggplot2)
library(tidyverse)
library(pheatmap)
library(viridisLite)
library(dendextend)
library(gprofiler2)
library(Seurat)
library(lme4)
library(ggpubr)

getnames = function(x){return(geneinfo[x,"hgnc"])}

```


```{r}
options(stringsAsFactors = F)
set.seed(157434538)
reanalyze=F
cutoff = 0.05 # pavalue cutoff for DEG
padj_cu=FALSE # use adjusted pvalue 
Dark8 = RColorBrewer::brewer.pal(8, "Dark2")
```

```{r getdata}
output= paste0("output/06_Bulk_DEX/")
dir.create(output, showWarnings = F)
source("code/custom_functions.R")
ddsMat <- readRDS("data/Bulk_dds_matrix.rds")
ddsMat <- ddsMat[,ddsMat$PAmM==0]


chr16p11_genes <- openxlsx::read.xlsx("data/chr16p11_orig.xlsx")
Kynurenine_genes <- openxlsx::read.xlsx("data/Kynurenine_genes_orig.xlsx")
Astrocyte_genes <- openxlsx::read.xlsx("data/AstrocyteGenes.xlsx")

SampleInfo = as.data.frame(colData(ddsMat))


log_2cpm=log2(counts(ddsMat, normalize=T)+1)

hgnc=gconvert(query=as.numeric(rownames(ddsMat)), 
              organism = "hsapiens", 
              numeric_ns = "ENTREZGENE_ACC",
              target = "HGNC")



```

# Differential gene expression 

```{r analyze DEX, eval=FALSE}
## comparisons against 0MM PA
PA=levels(SampleInfo$PAmM)
Days=c(sort(unique(SampleInfo$Day)))

p = 0
d = Days[1]
target = "PAmM"
randomeffect = "Batch"

geneinfo = rowData(ddsMat) %>% as.data.frame()
geneinfo$entrez_gene = rownames(geneinfo)
geneinfo$hgnc[is.na(geneinfo$hgnc)] <- geneinfo$entrez_gene[is.na(geneinfo$hgnc)]

for(d in Days){
  Dayfilter = SampleInfo$Day == d
  Set = rownames(SampleInfo)[Dayfilter]
  lab = paste("restab", "Day", d,"0vs5mM_PA",sep="_")
  print(lab)
  res <- comparison(dds_object = ddsMat, 
                    samples = Set, 
                    target = target, 
                    randomeffect = randomeffect)
  res$HGNC = geneinfo[match(as.character(rownames(res)), as.character(rownames(geneinfo))), "hgnc"]
  assign(lab, res)
}

#Lists elements are written to individual worksheets, using list names as sheet names if available
l <- list(base::mget(apropos("restab_")))[[1]]
names(l) = gsub("restab_", "", names(l)) %>% gsub("Day_5", "Day_05",.)
l = l[sort(names(l))]
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_DEX_LFC_results.xlsx"), row_names=T )

```


# Differential gene expression between timepoints 0mM only 

```{r analyze DEX TP}
## comparisons against 0MM PA
PA=levels(SampleInfo$PAmM)
Days=c(sort(unique(SampleInfo$Day)))

p = 0
d = Days[1]
target = "Day"
randomeffect = "Batch"

geneinfo = rowData(ddsMat) %>% as.data.frame()
geneinfo$entrez_gene = rownames(geneinfo)
geneinfo$hgnc[is.na(geneinfo$hgnc)] <- geneinfo$entrez_gene[is.na(geneinfo$hgnc)]

geneuniverses <- geneinfo$hgnc

for(i in 1:length(Days)){
  if(i==1){} else {
    Dayfilter = SampleInfo$Day %in% Days[c(i-1,i)] & SampleInfo$PAmM==0
    Set = rownames(SampleInfo)[Dayfilter]
    lab = paste("TPrestab", "Day", paste0(Days[c(i-1,i)], collapse = "vs"),"0mM_PA",sep="_")
    print(lab)
    res <- comparison(dds_object = ddsMat, 
                      samples = Set, 
                      target = target,
                      randomeffect = randomeffect)
    res$HGNC = rownames(res) %>% getnames()
    assign(lab, res)
  }
}

#Lists elements are written to individual worksheets, using list names as sheet names if available
l <- list(mget(apropos("TPrestab_")))[[1]]
names(l) = gsub("TPrestab_", "", names(l)) %>% gsub("Day_5", "Day_05",.)
l = l[sort(names(l))]
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_timpoints_DEX_LFC_results.xlsx"), row_names=T )




## TODO PLOT Manhatten regulated genes 
((TPrestab_Day_5vs10_0mM_PA$adj.P.Val<0.05) & (TPrestab_Day_5vs10_0mM_PA$logFC<0))*1 %>% sum()
(TPrestab_Day_10vs17_0mM_PA$adj.P.Val<0.05) %>% sum()
(TPrestab_Day_17vs40_0mM_PA$adj.P.Val<0.05) %>% sum()
(TPrestab_Day_40vs80_0mM_PA$adj.P.Val<0.05) %>% sum()





```

## GO terms comparing eacht timpoint gainst previous timepoint Wildtype only

```{r, fig.width=15, fig.height=20}

GOtermlist <- list()

for (i in apropos("TPrestab_")){
  restab <- get(i)
  if(padj_cu){
    GOtermlist[[paste0(i, "_all")]] <- 
      rownames(restab)[restab$adj.P.Val < cutoff] %>% getnames()
    GOtermlist[[paste0(i, "_up")]] <- 
      rownames(restab)[restab$adj.P.Val < cutoff  & restab$logFC >0]%>% getnames()
    GOtermlist[[paste0(i, "_dwn")]] <- 
      rownames(restab)[restab$adj.P.Val < cutoff  & restab$logFC < 0]%>% getnames()
  }else{
    GOtermlist[[paste0(i, "_all")]] <- 
      rownames(restab)[restab$P.Value < cutoff]%>% getnames()
    GOtermlist[[paste0(i, "_up")]] <- 
      rownames(restab)[restab$P.Value < cutoff  & restab$logFC >0]%>% getnames()
    GOtermlist[[paste0(i, "_dwn")]] <- 
      rownames(restab)[restab$P.Value < cutoff  & restab$logFC < 0]%>% getnames()
  }
}



names(GOtermlist) = gsub("TPrestab_Day", "Day", names(GOtermlist))
names(GOtermlist) = gsub("Day_5", "Day_05", names(GOtermlist))
GOtermlist = GOtermlist[sort(names(GOtermlist))]



resGO = getGOresults(GOtermlist, genereference = NULL, evcodes = T)

openxlsx::write.xlsx(file = paste0(output, "/06_Bulk_Timepoints_DEX_GO_results.xlsx"), as.data.frame(resGO$result))


Sfari_Syndromic_test <- test_Genelist(mygenelist = GOtermlist, 
                                      basegenelist = SFARI$gene.symbol[SFARI$syndromic==1], 
                                      geneunivers = geneuniverses)



Sfari_All_test <- test_Genelist(mygenelist = GOtermlist, 
                                basegenelist = SFARI$gene.symbol, 
                                geneunivers = geneuniverses)



l = list(Sfari_all=Sfari_All_test, Sfari_Syndromic=Sfari_Syndromic_test)
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_Timepoints_DEX_SFARIenrichment.xlsx"), row_names=T)


```


```{r, fig.width=15, fig.height=20}

plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_Timepoints_DEX_top10all_GOplot.svg"), plot = p)
```

```{r, fig.width=15, fig.height=20}
plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set & resGO$result$source == "GO:BP"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_Timepoints_DEX_top10_BP_GOplot.svg"), plot = p)
```


```{r, fig.width=15, fig.height=20}
plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set & resGO$result$source == "TF"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_Timepoints_DEX_top10_TF_GOplot.svg"), plot = p)
```


# Differential gene expression time x PA

```{r analyze DEX TPxPA, eval=FALSE}
## comparisons against 0MM PA
PA=SampleInfo$PAmM
Days=SampleInfo$Day
Batch=SampleInfo$Batch
p = 0
d = Days[1]
target = "Day*PAmM"
randomeffect = "Batch"

geneinfo = rowData(ddsMat) %>% as.data.frame()
geneinfo$entrez_gene = rownames(geneinfo)
geneinfo$hgnc[is.na(geneinfo$hgnc)] <- geneinfo$entrez_gene[is.na(geneinfo$hgnc)]

lab = paste("Restab_TPxPA")
print(lab)
res <- comparison(dds_object = ddsMat, 
                  target = target,
                  randomeffect = randomeffect)
res$HGNC = rownames(res) %>% getnames()
assign(lab, res)

#Lists elements are written to individual worksheets, using list names as sheet names if available
l <- list(mget(apropos("Restab_TPxPA")))[[1]]
names(l) = "DayxPA"
l = l[sort(names(l))]
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_DayxPA_DEX_LFC_results.xlsx"), row_names=T )



results<-data.frame(matrix(ncol=2, nrow=0))
for(g in rownames(log_2cpm)){
  print(which(rownames(log_2cpm)==g))
  subdat<-data.frame(reads=log_2cpm[g,], Days, PA, Batch)
  res <- lme4::glmer(reads~Days*PA+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[4]))
}

colnames(results)=c("Entrez","P.value")
results$P.Adj <- p.adjust(results$P.value, method = "fdr")
results$HGNC <- hgnc$name[match(results$Entrez, hgnc$input)]
openxlsx2::write_xlsx(results, file=paste0(output, "06_Bulk_DayxPA_DEX_DxPspec.Pval_results.xlsx"), row_names=T)


```

## GO terms  Day xPA 

```{r, fig.width=15, fig.height=20, eval=F}

GOtermlist <- list()

restab <- Restab_TPxPA
i="Restab_TPxPA"
if(padj_cu){
  GOtermlist[[paste0(i, "_all")]] <- 
    rownames(restab)[restab$adj.P.Val < cutoff] %>% getnames()
  GOtermlist[[paste0(i, "_up")]] <- 
    rownames(restab)[restab$adj.P.Val < cutoff  & restab$Day.PAmM5 >0]%>% getnames()
  GOtermlist[[paste0(i, "_dwn")]] <- 
    rownames(restab)[restab$adj.P.Val < cutoff  & restab$Day.PAmM5 < 0]%>% getnames()
}else{
  GOtermlist[[paste0(i, "_all")]] <- 
    rownames(restab)[restab$P.Value < cutoff]%>% getnames()
  GOtermlist[[paste0(i, "_up")]] <- 
    rownames(restab)[restab$P.Value < cutoff  & restab$Day.PAmM5 >0]%>% getnames()
  GOtermlist[[paste0(i, "_dwn")]] <- 
    rownames(restab)[restab$P.Value < cutoff  & restab$Day.PAmM5 < 0]%>% getnames()
}


names(GOtermlist) = gsub("Restab_TPxPA", "TPxPA", names(GOtermlist))
GOtermlist = GOtermlist[sort(names(GOtermlist))]

resGO = getGOresults(GOtermlist, genereference = NULL, evcodes = T)

openxlsx::write.xlsx(file = paste0(output, "/06_Bulk_DayxPA_DEX_GO_results.xlsx"), as.data.frame(resGO$result))


Sfari_Syndromic_test <- test_Genelist(mygenelist = GOtermlist, 
                                      basegenelist = SFARI$gene.symbol[SFARI$syndromic==1], 
                                      geneunivers = geneuniverses)



Sfari_All_test <- test_Genelist(mygenelist = GOtermlist, 
                                basegenelist = SFARI$gene.symbol, 
                                geneunivers = geneuniverses)



l = list(Sfari_all=Sfari_All_test, Sfari_Syndromic=Sfari_Syndromic_test)
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_DayxPA_DEX_SFARIenrichment.xlsx"), row_names=T)


```


```{r, fig.width=15, fig.height=20, eval=F}

plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DayxPA_DEX_top10all_GOplot.svg"), plot = p)
```

```{r, fig.width=15, fig.height=20, eval=F}
plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set & resGO$result$source == "GO:BP"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DayxPA_DEX_top10_BP_GOplot.svg"), plot = p)
```


```{r, fig.width=15, fig.height=20, eval=F}
plotlist = list()
for (set in unique(resGO$result$query)){
  dat=resGO$result[which(resGO$result$query == set & resGO$result$source == "TF"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else{
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DayxPA_DEX_top10_TF_GOplot.svg"), plot = p)
```


# Trajectories over time top genes per comparison

```{r plot top 10 DEX genes, fig.width=15, fig.height=15}


#alle_restab <- apropos("restab_Day") # gibt alle wieder die restab beinhalten egal ob groß oder klein schreiben
#restab_nurTP0mM_und_Day0vs5 <- apropos("restab_", ignore.case = F) # berücksichtig groß und kleinschreibung, 
#restab_nur_Day0vs5  <- apropos("^restab_", ignore.case = F) # beginnt mit retab_ kleingeschreiben
restab_nur_TP0mM  <- apropos("TPrestab_", ignore.case = F) # Beinhaltet TPrestab


topgenes=c()
for(r in restab_nur_TP0mM){
  res <- get(r)
  topgenes <- c(topgenes, res$HGNC[1:10])
}

topgenes <- unique(topgenes)
topgenes <- sort(topgenes)

my_comparisons <- list(c(5, 10), c(10, 17), c(14, 40), c(40,80))


plotDEX= function(geneToPlot){
  data.frame(reads = log_2cpm[which(geneinfo$hgnc==geneToPlot),],
             Day = as.factor(SampleInfo$Day)) -> Plotdata
  Plotdata %>% ggboxplot(x = "Day", y = "reads", color="Day", add="jitter")+
    theme_classic() +
    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
    labs(x = "Day", y = "log2(CPM+1)", title = geneToPlot) -> p
  return(p)
}


plotlist=list()
for(g in topgenes){
  p = plotDEX(g)
  plotlist[[g]] <- p
}

pcomb = ggpubr::ggarrange(plotlist=plotlist, nrow = 5, ncol = 4, common.legend = T, heights = c(3,3,3,3,3), widths = c(3,3,3,3))
pcomb
ggpubr::ggexport(filename = paste0(output, "/06_Bulk_DEX_Siggenes_boxplots.pdf"), plotlist = pcomb, width = 14, height = 21, dpi = 300)

dataplot <- log_2cpm[which(geneinfo$hgnc%in%topgenes),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads)) +
  geom_smooth(span=0.5) +geom_jitter()+ scale_x_continuous(breaks=c(5,10,17,40,80))+
  theme_classic() + facet_wrap(~hgnc) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_Siggenes_lineplots.svg"), p)
```

## 16p11 genes over time

```{r, fig.width=7, fig.height=10}
dataplot <- log_2cpm[which(geneinfo$hgnc%in%chr16p11_genes$Gene_symbol),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$Batch = SampleInfo[dataplot$Sample, "Batch"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads)) +
  geom_smooth(span=0.5) +geom_jitter()+ scale_x_continuous(breaks=c(5,10,17,40,80))+
  theme_classic() + facet_wrap(~hgnc) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_chr16p11_lineplots.svg"), p)

results<-data.frame(matrix(ncol=2, nrow=0))

for(g in unique(dataplot$hgnc)){
  subdat<-dataplot[dataplot$hgnc == g, ]
  res <- lme4::glmer(reads~poly(Day,3)+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[2]))
}

colnames(results) <- c("gene", "Day.Pvalue")

openxlsx2::write_xlsx(file=paste0(output, "/06_Bulk_DEX_chr16p11_stat_Day.xlsx"), results)

colnames(results) <- c("gene", "Pval")
display_tab(results)


```


## Kynurenine genes over time


```{r, fig.width=7, fig.height=10}
dataplot <- log_2cpm[which(geneinfo$hgnc%in%Kynurenine_genes$Gene_symbol),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$PAmM = SampleInfo[dataplot$Sample, "PAmM"]
dataplot$Batch = SampleInfo[dataplot$Sample, "Batch"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads)) +
  geom_smooth(span=0.5) + geom_jitter()+ scale_x_continuous(breaks=sort(unique(dataplot$Day)))+
  theme_classic() + facet_wrap(~hgnc) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_Kynurenine_lineplots.svg"), p)

results<-data.frame(matrix(ncol=2, nrow=0))
for(g in unique(dataplot$hgnc)){
  subdat<-dataplot[dataplot$hgnc == g, ]
  res <- lme4::glmer(reads~poly(Day,3)+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[2]))
}

colnames(results) <- c("gene", "Day.Pvalue")

openxlsx2::write_xlsx(file=paste0(output, "/06_Bulk_DEX_Kynurenine_stat_Day.xlsx"), results)


display_tab(results)


```

## Astrocyte genes over time


```{r, fig.width=7, fig.height=10}
dataplot <- log_2cpm[which(geneinfo$hgnc%in%Astrocyte_genes$Gene_symbol),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$PAmM = SampleInfo[dataplot$Sample, "PAmM"]
dataplot$Batch = SampleInfo[dataplot$Sample, "Batch"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads)) +
  geom_smooth(span=0.5) +scale_x_continuous(breaks=sort(unique(dataplot$Day)))+
  theme_classic() + facet_wrap(~hgnc) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, "/06_Bulk_DEX_Astrocytes_lineplots.svg"), p)

results<-data.frame(matrix(ncol=2, nrow=0))
for(g in unique(dataplot$hgnc)){
  subdat<-dataplot[dataplot$hgnc == g, ]
  res <- lme4::glmer(reads~poly(Day,3)+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[2]))
}


colnames(results) <- c("gene", "Day.Pvalue")
openxlsx2::write_xlsx(file=paste0(output, "/06_Bulk_DEX_Astrocyte_stat_DayxmM.xlsx"), results)

display_tab(results)

```

## Selected genes over time


```{r, fig.width=7, fig.height=10}
targetgenes = c("SYN1", "CAMK2A", "LHX2", "SLITRK5")
plotoutputname<- "/06_Bulk_DEX_SelectedGenes_lineplots.svg"
xlsxoutputname <- "/06_Bulk_DEX_SelectedGenes_lineplots.xlsx"

dataplot <- log_2cpm[which(geneinfo$hgnc%in% targetgenes),] %>% reshape2::melt()
colnames(dataplot) <- c("gene","Sample", "reads")  
dataplot$PAmM = SampleInfo[dataplot$Sample, "PAmM"]
dataplot$Batch = SampleInfo[dataplot$Sample, "Batch"]
dataplot$Day = SampleInfo[dataplot$Sample, "Day"]
dataplot$hgnc = hgnc$target[match(dataplot$gene, hgnc$input)]
idx=is.na(dataplot$hgnc)
dataplot$hgnc[idx] = dataplot$gene[idx]

ggplot(dataplot, aes(x = Day, y = reads)) +
  geom_smooth(span=0.5) +scale_x_continuous(breaks=sort(unique(dataplot$Day)))+
  theme_classic() + facet_wrap(~hgnc) +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
  labs(x = "Day", y = "log2(CPM+1)") -> p

p

ggsave(file=paste0(output, plotoutputname), p)

results<-data.frame(matrix(ncol=2, nrow=0))
for(g in unique(dataplot$hgnc)){
  subdat<-dataplot[dataplot$hgnc == g, ]
  res <- lme4::glmer(reads~poly(Day,3)+(1|Batch), data = subdat)
  res_stat <- car::Anova(res, type="3")
  results<-rbind(results, c(g, res_stat$`Pr(>Chisq)`[2]))
}


colnames(results) <- c("gene", "Day.Pvalue")

openxlsx2::write_xlsx(file=paste0(output,xlsxoutputname), results)

colnames(results) <- c("gene", "Pval")
display_tab(results)

```

# WGCNA 0mM PA only (WT only)


```{r fig.width=8, fig.height=6}
library(org.Hs.eg.db)
library(WGCNA)

force.recalc = T
powers = c(seq(1,10,by=1), seq(12,20, by=2));

datExpr <- t(counts(ddsMat[,ddsMat@colData$PAmM=="0"], normalized=T))


# expressed in more than 10% of samples 
exprgenes = colSums(datExpr>0)>nrow(datExpr)*0.1
datExpr <- datExpr[,exprgenes]
hsacc=as.list(org.Hs.egACCNUM)
annotatedgenes = colnames(datExpr) %in% names(hsacc)
datExpr <- datExpr[,annotatedgenes]

multiExpr <- list()
multiExpr[['ODC']] <- list(data=datExpr)

checkSets(multiExpr) # check data size


if(force.recalc | (! file.exists(file=paste0(output,"/net.rds")))){ # skips recalculation of net object as it takes long... 
  
  
  
  #annotated only 
  
  
  
  
  powers = c(seq(1,10,by=1), seq(12,20, by=2));
  
  
  enableWGCNAThreads()
  
  # Call the network topology analysis function for each set in turn
  powerTable = list(
    data = pickSoftThreshold(
      datExpr,
      powerVector=powers,
      verbose = 100,
      networkType="unsigned",
      corFnc="cor"
    )[[2]]
  );
  
  
  colors = c("blue", "red","black")
  # Will plot these columns of the returned scale free analysis tables
  plotCols = c(2,5,6,7)
  colNames = c("Scale Free Topology Model Fit", "Mean connectivity", "mean connectivity",
               "Max connectivity");
  
  # Get the minima and maxima of the plotted points
  ylim = matrix(NA, nrow = 2, ncol = 4);
  for (col in 1:length(plotCols)){
    ylim[1, col] = min(ylim[1, col], powerTable$data[, plotCols[col]], na.rm = TRUE);
    ylim[2, col] = max(ylim[2, col], powerTable$data[, plotCols[col]], na.rm = TRUE);
  }
  
  # Plot the quantities in the chosen columns vs. the soft thresholding power
  
  par(mfcol = c(2,2));
  par(mar = c(4.2, 4.2 , 2.2, 0.5))
  cex1 = 0.7;
  
  for (col in 1:length(plotCols)){
    plot(powerTable$data[,1], -sign(powerTable$data[,3])*powerTable$data[,2],
         xlab="Soft Threshold (power)",ylab=colNames[col],type="n", ylim = ylim[, col],
         main = colNames[col]);
    addGrid();
    
    if (col==1){
      text(powerTable$data[,1], -sign(powerTable$data[,3])*powerTable$data[,2],
           labels=powers,cex=cex1,col=colors[1]);
    } else
      text(powerTable$data[,1], powerTable$data[,plotCols[col]],
           labels=powers,cex=cex1,col=colors[1]);
    if (col==1){
      legend("bottomright", legend = 'Metacells', col = colors, pch = 20) ;
    } else
      legend("topright", legend = 'Metacells', col = colors, pch = 20) ;
  }
  
  softPower=4
  
  nSets = 1
  setLabels = 'OCS'
  shortLabels = setLabels
  
  
  
  # construct network
  
  net=blockwiseConsensusModules(multiExpr, blocks = NULL,
                                maxBlockSize = 30000, ## This should be set to a smaller size if the user has limited RAM
                                randomSeed = 12345,
                                corType = "pearson",
                                power = softPower,
                                consensusQuantile = 0.3,
                                networkType = "unsigned",
                                TOMType = "signed",
                                TOMDenom = "min",
                                scaleTOMs = TRUE, scaleQuantile = 0.8,
                                sampleForScaling = TRUE, sampleForScalingFactor = 1000,
                                useDiskCache = TRUE, chunkSize = NULL,
                                deepSplit = 4,
                                pamStage=FALSE,
                                detectCutHeight = 0.995, minModuleSize = 50,
                                mergeCutHeight = 0.2,
                                saveConsensusTOMs = TRUE,
                                consensusTOMFilePattern = "ConsensusTOM-block.%b.rda")
  
  saveRDS(net, file=paste0(output,"net.rds"))
  saveRDS(powerTable, file=paste0(output,"powertable.rds"))
} else {
  net = readRDS(file=paste0(output,"net.rds"))
  powerTable = readRDS(file=paste0(output,"powertable.rds"))
  powers = c(seq(1,10,by=1), seq(12,20, by=2));
  colors = c("blue", "red","black")
  # Will plot these columns of the returned scale free analysis tables
  plotCols = c(2,5,6,7)
  colNames = c("Scale Free Topology Model Fit", "Mean connectivity", "mean connectivity",
               "Max connectivity");
  
  # Get the minima and maxima of the plotted points
  ylim = matrix(NA, nrow = 2, ncol = 4);
  for (col in 1:length(plotCols)){
    ylim[1, col] = min(ylim[1, col], powerTable$data[, plotCols[col]], na.rm = TRUE);
    ylim[2, col] = max(ylim[2, col], powerTable$data[, plotCols[col]], na.rm = TRUE);
  }
  
  # Plot the quantities in the chosen columns vs. the soft thresholding power
  
  par(mfcol = c(2,2));
  par(mar = c(4.2, 4.2 , 2.2, 0.5))
  cex1 = 0.7;
  
  for (col in 1:length(plotCols)){
    plot(powerTable$data[,1], -sign(powerTable$data[,3])*powerTable$data[,2],
         xlab="Soft Threshold (power)",ylab=colNames[col],type="n", ylim = ylim[, col],
         main = colNames[col]);
    addGrid();
    
    if (col==1){
      text(powerTable$data[,1], -sign(powerTable$data[,3])*powerTable$data[,2],
           labels=powers,cex=cex1,col=colors[1]);
    } else
      text(powerTable$data[,1], powerTable$data[,plotCols[col]],
           labels=powers,cex=cex1,col=colors[1]);
    if (col==1){
      legend("bottomright", legend = 'Metacells', col = colors, pch = 20) ;
    } else
      legend("topright", legend = 'Metacells', col = colors, pch = 20) ;
  }
}

consMEs = net$multiMEs;
moduleLabels = net$colors;


# Convert the numeric labels to color labels
moduleColors = as.character(moduleLabels)
consTree = net$dendrograms[[1]];



# module eigengenes
MEs=moduleEigengenes(multiExpr[[1]]$data, colors = moduleColors, nPC=1)$eigengenes
MEs=orderMEs(MEs)
meInfo<-data.frame(rownames(datExpr), MEs)
colnames(meInfo)[1]= "SampleID"

# intramodular connectivity
KMEs<-signedKME(datExpr, MEs,outputColumnName = "kME",corFnc = "bicor")

# compile into a module metadata table
geneInfo=as.data.frame(cbind(colnames(datExpr),moduleColors, KMEs))

# how many modules did we get?
nmodules <- length(unique(moduleColors))

# merged gene symbol column
colnames(geneInfo)[1]= "GeneSymbol"
colnames(geneInfo)[2]= "Initially.Assigned.Module.Color"
geneInfo$HGNC=hgnc$name[match(geneInfo$GeneSymbol, hgnc$input)]
geneInfo$SFARI_genescore <- SFARI$gene.score[match(geneInfo$HGNC,SFARI$gene.symbol)]
geneInfo$SFARI_category <- SFARI$genetic.category[match(geneInfo$HGNC,SFARI$gene.symbol)]

# save info


PCvalues=MEs
```

## merge modules

```{r fig.width=8, fig.height=12}
par(cex = 0.9)
plotEigengeneNetworks(PCvalues, "", 
                      marDendro = c(0,4,1,4), 
                      marHeatmap = c(4,5,1,2), 
                      cex.lab = 0.8, 
                      xLabelsAngle= 90)

MEDissThres = 0.5


merged_col<- mergeCloseModules(datExpr,moduleColors, 
                               cutHeight = MEDissThres)

ME <- as.data.frame(merged_col$newMEs)

plotEigengeneNetworks(ME, "", 
                      marDendro = c(0,4,1,4), 
                      marHeatmap = c(4,5,1,2), 
                      cex.lab = 0.8, 
                      xLabelsAngle= 90)


WGCNA::plotDendroAndColors(consTree, merged_col$colors, "Module colors", 
                               dendroLabels = FALSE, hang = 0.03, 
                               addGuide = TRUE, guideHang = 0.05,
                               main = paste0("dendrogram and module colors"))


svg(paste0(output, "06_Bulk_DEX_WGCNA_Dendrocolplot.svg"))
WGCNA::plotDendroAndColors(consTree, merged_col$colors, "Module colors", 
                               dendroLabels = FALSE, hang = 0.03, 
                               addGuide = TRUE, guideHang = 0.05,
                               main = paste0("dendrogram and module colors"))

dev.off()


geneInfo$Module_color <- merged_col$colors


openxlsx2::write_xlsx(geneInfo %>% 
                        dplyr::select(GeneSymbol, HGNC,SFARI_genescore, SFARI_category, Module_color), 
                      file=paste0(output,'/GeneInfoSigned.xlsx'))

```



```{r, fig.width=12, fig.height=12}
## Todo adjust 

PCvalues = ME %>%  dplyr::select(-MEgrey)
targets=colData(ddsMat[,ddsMat@colData$PAmM==0]) %>% as.data.frame()

targes = targets[rownames(PCvalues),]
PCvalues_norm=t(limma::removeBatchEffect(t(PCvalues), batch=targets$Batch))


plot_df <- cbind(dplyr::select(targets, c(Day, Batch)), PCvalues_norm)
plot_df <- reshape2::melt(plot_df, id.vars = c('Day', "Batch"))

colors <- sub('ME', '', as.character(levels(plot_df$variable)))
names(colors)<- as.character(levels(plot_df$variable))


p <- ggplot(plot_df, aes(x=Day, y=value, fill=variable, color=variable)) + 
  geom_point(color="black")+
  stat_summary(fun.y = mean, geom = 'smooth', color="black") + 
  scale_x_continuous(breaks=sort(unique(plot_df$Day)))+
  RotatedAxis() + 
  ylab('Module Eigengene') + 
  xlab('') +
  theme(
    axis.text.x=element_blank(),
    axis.ticks.x=element_blank(),
  )+scale_fill_manual(values = colors)+
  scale_color_manual(values = colors)

p <- p + facet_wrap(~variable, scales="fixed")
#p<- p + stat_summary(fun.data = 'mean_sdl', mult = 1, geom = 'ribbon', alpha=0.5)

p <- p + geom_smooth(span=0.5, lwd=2)
p

ggsave(filename = paste0(output, "/06_Bulk_DEX_WGCNA_plots.svg"), p)

```



```{r, fig.width=6, fig.height=4}

reslist = list()


for(M in unique(plot_df$variable)){
  
  Mdata = plot_df[plot_df$variable==M,]
  fit=lmer(value~poly(Day,3)+(1|Batch), data = Mdata)
  resfit = car::Anova(fit, type=3)%>% as.data.frame()
  res = c(Day.pval = resfit["poly(Day, 3)", "Pr(>Chisq)"])
  
  reslist[[M]] <- res
  
}

wgcnastats <- do.call(rbind, reslist) %>% as.data.frame()

rownames(wgcnastats) <- names(reslist)
wgcnastats$Day.adjp <- p.adjust(wgcnastats$Day.pval, method = "b")

openxlsx2::write_xlsx(wgcnastats, paste0(output, "06_DEX_Bulk_WGCNA_Stats.xlsx"),
                      row_names=T)

```


```{r, fig.width=12, fig.height=12}
targets$Day_PA = paste0(targets$Day, "d_",targets$PAmM, "mMPA")
targets$Day_PA <- factor(targets$Day_PA, levels=c("5d_0mMPA", "5d_5mMPA", "10d_0mMPA","10d_5mMPA", "17d_0mMPA", "17d_5mMPA","40d_0mMPA", "40d_5mMPA","80d_0mMPA",   "80d_5mMPA"))

MEs<- ME

Meandata = MEs %>% group_by(targets$Day_PA) %>% summarise_all(mean) %>% column_to_rownames("targets$Day_PA") %>% t()
SDdata = MEs %>% group_by(targets$Day_PA) %>% summarise_all(sd)%>% column_to_rownames("targets$Day_PA") %>% t()
colLabels=colnames(Meandata)
rowLabels=rownames(Meandata)
textlab = paste0(signif(Meandata,2), "\n", "(", signif(SDdata, 2), ")")

svg(paste0(output,"/06_DEX_WGCNA_labeled_Heatmap.svg"))
labeledHeatmap(Meandata, xLabels = colLabels, 
               yLabels = rowLabels,
               colorLabels = TRUE,
               colors = viridis(50),
               setStdMargins = FALSE, textMatrix = textlab,
               cex.col="white",
               main = "Mean(SD) Eigengene-expression");
dev.off()
```

## WGCNA GO terms

```{r, fig.width=15, fig.height=20}

net <- merged_col
genelist = split(rownames(ddsMat)%>% getnames(), net$colors) 
gene_univers = rownames(ddsMat)%>% getnames()


names(genelist)

pvallist = unlist(reslist)
MOI = unique(net$colors)

# if you want to have the affected genes set return_genelist = T; this takes roughly 4-8 hours 

Gores = getGOresults(geneset = genelist[MOI], evcodes = T,
                     #genereference = gene_univers, 
                     organism = "hsapiens")

display_tab(Gores$result)

restab = Gores$result

restab$parents<- as.character(restab$parents)

openxlsx2::write_xlsx(restab, file=paste0(output,"06_DEX_Bulk_WGCNA_GO_Terms.xlsx"))



Sfari_Syndromic_test <- test_Genelist(mygenelist = genelist[MOI], 
                                      basegenelist = SFARI$gene.symbol[SFARI$syndromic==1], 
                                      geneunivers = gene_univers)



Sfari_All_test <- test_Genelist(mygenelist = genelist[MOI], 
                                basegenelist = SFARI$gene.symbol, 
                                geneunivers = gene_univers)



l = list(Sfari_all=Sfari_All_test, Sfari_Syndromic=Sfari_Syndromic_test)
openxlsx2::write_xlsx(l, file=paste0(output, "06_Bulk_WGCNA_SFARIenrichment.xlsx"), row_names=T)


plotlist = list()
for (set in unique(Gores$result$query)){
  dat=Gores$result[which(Gores$result$query == set & Gores$result$source == "GO:BP"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else {
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DEX_WGCNA_BP_GOplot.svg"), plot = p)


plotlist = list()
for (set in unique(Gores$result$query)){
  dat=Gores$result[which(Gores$result$query == set & Gores$result$source == "TF"),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else {
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DEX_WGCNA_TF_GOplot.svg"), plot = p)


plotlist = list()
for (set in unique(Gores$result$query)){
  dat=Gores$result[which(Gores$result$query == set),]
  if(nrow(dat)>0){
    plotlist[[set]]<-GOplot(dat, N = 10, Title = set)
  }else {
    text="no significance"
    plotlist[[set]]<-ggplot() + annotate("text", x = 4, y = 25, label = text) + 
      ggtitle(set) + theme_void()
  }
}

p <- ggpubr::ggarrange(plotlist = plotlist, common.legend = T, nrow = 5, ncol =3)
p
ggsave(file=paste0(output, "/06_Bulk_DEX_WGCNA_GOplot.svg"), plot = p)

```

